packages/core/src/application.ts

```diff
@@ -216,6 +222,26 @@ export class Application extends Context {
   public setMetadata(metadata: ApplicationMetadata) {
     this.bind(CoreBindings.APPLICATION_METADATA).to(metadata);
   }
+
+  /**
+   * Register a life cycle observer class
+   * @param ctor A class implements LifeCycleObserver
+   * @param name Optional name for the life cycle observer
+   */
+  public lifeCycleObserver<T extends LifeCycleObserver>(
+    ctor: Constructor<T>,
+    name?: string,
+  ): Binding<T> {
+    debug('Adding life cycle observer %s', name || ctor.name);
+    const binding = createBindingFromClass(ctor, {
+      name,
+      namespace: CoreBindings.LIFE_CYCLE_OBSERVERS,
+      type: CoreTags.LIFE_CYCLE_OBSERVER,
+      defaultScope: BindingScope.SINGLETON,
+    }).apply(asLifeCycleObserver);
```

Comment:
Can we move the default namespace, type and defaultScope to `asLifeCycleObserver` template?

```ts
const binding = createBindingFromClass(ctor, {name})
  .apply(asLifeCycleObserver);
```

Reply:
Or even better:

```ts
app.bindFromClass(ctor, {name}, asLifeCycleObserver);
```

Reply:
> Can we move the default namespace, type and defaultScope to asLifeCycleObserver template?

In fact, no. Such information is used to infer binding attributes but the template function is applied after the binding is created.

Reply:
Does it mean that if the lifecycle class is decorated with `@bind` and sets the type to something else than `CoreTags.LIFE_CYCLE_OBSERVER`, then the result of calling `app.lifeCycleObserver` is a binding with a type that's not LIFE_CYCLE_OBSERVER? Similarly for `namespace`. Are we okay with that? I find it surprising to be honest.